lib/Pingmachine/Probe/HTTPing.pm

```diff
@@ -0,0 +1,256 @@
+);
+
+with 'Pingmachine::Probe';
+
+sub _start_new_job {
+    my ($self) = @_;
+    my $cv;
+
+    my $step  = $self->step;
+    my $pings = $self->pings;
+    my $interval = $self->interval;
+    my $user_agent = $self->user_agent;
+
+    my $urlcount = $self->order_list->count;
+
+    return unless $urlcount;
+
+    # Make sure that we can process the request
+    if(($TIMEOUT + $interval * 1000) * $pings > $step * 1000) {
+        die "httping: step * 1000 must be higher than (timeout + interval * 1000) * pings (step = $step, timeout=$TIMEOUT, interval=$interval, pings=$pings)\n";
+    }
+
+    # Prepare job
+    my %job = (
+        url2order  => {},
+        output     => {},
+        cmd        => {},
+        pid        => {},
+    );
+    for my $order ($self->order_list->get_all) {
+        my $url = $order->httping->url;
+        push @{$job{url2order}{$url}}, $order;
+    }
+    $self->current_job(\%job);
+    
+    for my $url (keys %{$job{url2order}}) {
+        # Run httping
+        my $cmd = [
+            $HTTPING_BIN,
+            $url,
+            '-I', $user_agent,
+            '-c', $pings,
+            '-i', $interval,
+            '-t', $TIMEOUT,
+        ];
+
+        if ( $self->proxy ) {
+            push @{$cmd}, '-x';
+            push @{$cmd}, $self->proxy;
+        }
+
+        $job{cmd}{$url} = join(' ', @$cmd);
+        $cv = run_cmd $cmd,
+            '2>', '/dev/null',
+            '>', \$job{output}{$url},
+            '$$', \$job{pid}{$url};
+    }
+
+    $cv->cb(
+        sub {
+            my $cbv = shift;
+            for my $url (keys %{$job{url2order}}) {
+                $job{pid}{$url} = undef;
+            }
+            my $exit = $cbv->recv;
+            $exit = $exit >> 8;
+            if($exit and $exit != 1 and $exit != 2) {
+                # exit 1 means that some urls aren't reachable
+                # exit 2 means "any IP addresses were not found"
+                $log->warning("httping seems to have failed (exit: $exit, stderr: " . Dumper(\$job{output}) . ")");
+                return;
+            }  
+
+            $log->debug("finished:" . Dumper(\$job{cmd}) . "(step: $step, pings: $pings, offset: " . $self->time_offset() . ")") if $log->is_debug();
+                
+            $self->_collect_current_job();
+
+            $log->debug("collected:". Dumper(\$job{cmd}) . "(step: $step, pings: $pings, offset: " . $self->time_offset() . ")") if $log->is_debug();
+        }
+    );
+}
+
+sub _kill_current_job {
+    my ($self) = @_;
+
+    # Kill httping, if still running
+    my $job = $self->current_job;
+    for my $url (keys %{$job->{url2order}}) {
+        if($job->{pid}{$url}) {
+            # Check that we are killing the process we started and not an innocent bystander
+            my $cmd_match = 0;
+            if (open(proc_fh, "/proc/$job->{pid}{$url}/cmdline")) {
```

Comment:
please do not use bareword file handles (better `my $proc_fh` instead of `proc_fh`)